src/site/_includes/components/CodePattern.js

```diff
@@ -0,0 +1,66 @@
+const {html} = require('common-tags');
+const Prism = require('prismjs');
+
+module.exports = () => {
+  const pattern = {
+    id: 'example-pattern',
+    layout: 'pattern',
+    title: 'An example pattern in a suite',
+    description: 'A description for the example pattern',
+    content: '<h2>An example pattern in a suites</h2>',
+    assets: {
+      'body.html': {
+        content: '<h1>Hello</h1>',
+        type: 'html',
+      },
+      'script.js': {
+        content: 'document.write("hello")',
+        type: 'js',
+      },
+      'style.css': {
+        content: 'background: red;',
+        type: 'css',
+      },
+    },
+    demo: '/patterns/example-pattern/demo.html',
+    suite: '/patterns/',
+  };
+
+  const assets = Object.values(pattern.assets)
+    .map((asset) => {
+      const content = Prism.highlight(
+        asset.content,
+        Prism.languages.html,
+        'html',
+      );
+      return html`
+        <web-tab title="${asset.type}">
+          <pre>
+            <code class="language-${asset.type}">${content}</code>
```

Comment:
Can you try a multiline code example. My concern is that `pre` preserves whitespace, so it seems like the way this is formatted will lead to weird indentation.

Reply:
It looks like the pre element is including the whitespace, the first line of html is shifted to the right a bit.

![image](https://user-images.githubusercontent.com/1066253/128762760-2d083702-4dd8-4636-8b6b-99e60a448c3d.png)